apollo-router/src/plugins/telemetry/tracing/apollo_telemetry.rs

```diff
@@ -523,16 +538,39 @@ fn extract_i64(v: &Value) -> Option<i64> {
     }
 }
 
-fn extract_ftv1_trace(v: &Value) -> Option<Result<Box<proto::reports::Trace>, Error>> {
+fn extract_ftv1_trace(
+    v: &Value,
+    error_config: &ErrorConfiguration,
+) -> Option<Result<Box<proto::reports::Trace>, Error>> {
     if let Value::String(s) = v {
-        if let Some(t) = decode_ftv1_trace(s.as_str()) {
+        if let Some(mut t) = decode_ftv1_trace(s.as_str()) {
+            if error_config.redact || !error_config.send {
+                if let Some(root) = &mut t.root {
+                    redact_node_errors(root, !error_config.send);
```

Comment:
I feel this code is confusing. The function is called `redact_node_errors` but takes the send config.
It'd be easier to get if it had the following signature `preprocess_errors(root, redact, send)`

Reply:
Agree with this feedback, but let's follow-up on this in another PR.

Reply:
Done here https://github.com/apollographql/router/pull/3030